aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -318,4 +264,158 @@ public Completable markInFlight(@NonNull TimeBasedUuid pendingMutationId) {
+                    return modelAlreadyScheduledForDeletion();
+                default:
+                    return unexpectedMutationScenario();
+            }
+        }
+
+        /**
+         * Determine which action to take when the incoming mutation type is {@linkplain PendingMutation.Type#DELETE}.
+         * @return A completable with the actions needed to resolve the conflict
+         */
+        private Completable handleIncomingDelete() {
+            switch (existing.getMutationType()) {
+                case CREATE:
+                    //
+                    if (inFlightMutations.contains(existing.getMutationId())) {
+                        // Existing create is already in flight, then save the delete
+                        return save(incoming);
+                    } else {
+                        // The existing create mutation hasn't made it to the remote store, so we
+                        // ignore the incoming and remove the existing create mutation from outbox.
+                        return remove(existing.getMutationId());
+                    }
+                case UPDATE:
+                case DELETE:
+                    // If there's a pending update OR delete, we want to replace it with the incoming delete.
+                    return overwriteExistingAndNotify(PendingMutation.Type.DELETE, incoming.getPredicate());
+                default:
+                    return unexpectedMutationScenario();
+            }
+        }
+
+        private Completable overwriteExistingAndNotify(@NonNull PendingMutation.Type type,
+                                                       @Nullable QueryPredicate predicate) {
+            // Keep the old mutation ID, but update the contents of that mutation.
+            // Now, it will have the contents of the incoming update mutation.
+            TimeBasedUuid id = existing.getMutationId();
+            T item = incoming.getMutatedItem();
+            Class<T> clazz = incoming.getClassOfMutatedItem();
+            return save(PendingMutation.instance(id, item, clazz, type, predicate))
+                .andThen(notifyContentAvailable());
+        }
+
+        private Completable saveIncomingAndNotify() {
+            return save(incoming)
+                .andThen(notifyContentAvailable());
+        }
+
+        private Completable conflictingCreationError() {
+            return Completable.error(new DataStoreException(
+                "Attempted to enqueue a model creation, but there is already a pending creation for that model ID.",
+                "Please report at https://github.com/aws-amplify/amplify-android/issues."
+            ));
+        }
+
+        private Completable modelAlreadyScheduledForDeletion() {
+            return Completable.error(new DataStoreException(
+                "Attempted to enqueue a model mutation, but that model already had a delete mutation pending.",
+                "This should not be possible. Please report on GitHub issues."
```

Comment:
Nit: Update suggestion to match others with a link - Jameson actually recently added a constant string to Amplify Exception for this type of suggestion you can use.

Reply:
Hm, it wasn't me, actually! I thought it was you. [It was @drochetti](https://github.com/aws-amplify/amplify-android/blame/d4e60a992eee495dab09ffc0a02bfa77776b1c1d/core/src/main/java/com/amplifyframework/AmplifyException.java#L33-L41)!